.hyperloop/checks/check-frontend-tests-exist.sh

```
@@ -16,11 +16,49 @@ set -euo pipefail
+  exit 0
+fi
+
+# Check whether there are any Vue/TS/JS source files in the UI directory.
+# Must match the same file-type set as task_ui_changes to avoid false negatives.
 ui_source_count=$(find "$UI_SOURCE_DIR" \
-  -name "*.vue" -o -name "*.ts" 2>/dev/null \
+  \( -name "*.vue" -o -name "*.ts" -o -name "*.js" \) 2>/dev/null \
   | grep -v node_modules \
-  | grep -v ".nuxt" \
+  | grep -v "\.nuxt" \
   | grep -v dist \
```

Shellcheck (0.11.0)

[style] Consider using 'grep -c' instead of 'grep|wc -l'.

(SC2126)